src/matlab/_utils.py

```diff
@@ -0,0 +1,58 @@
+import functools
+import os
+import platform
+import shutil
+from typing import Optional, NamedTuple
+
+
+def _get_platform_arch() -> str:
+    system_name = platform.system()
+    
+    if system_name == 'Windows':
+        return 'win64'
+    if system_name == 'Linux':
+        return 'glnxa64'
+    if system_name == 'Darwin':
+        if platform.mac_ver()[-1] == 'arm64':
+            return 'maca64'
+        return 'maci64'
+
+    raise RuntimeError(f"{system_name} is not a supported platform.")
+
+
+def _get_matlab_root() -> Optional[str]:
+    """Probe matlab root directory"""
+    matlab_command = shutil.which('matlab')
+    if not matlab_command:
+        return None
+    matlab_bin_dir = os.path.dirname(matlab_command)
+    matlab_root = os.path.normpath(os.path.join(matlab_bin_dir, os.pardir))
+    return matlab_root
+
+
+class MatlabPathInfo(NamedTuple):
+    arch: str
+    bin_folder: str
+    engine_folder: str
+    extern_bin: str
+
+
+@functools.cache
+def get_path_info() -> MatlabPathInfo:
```

Comment:
Read `_arch.txt` and use `which matlab` as fallback.
This could be more involved and use the whole `_MatlabFinder` logic.